### PR TITLE
#134 Setup formik with yup

### PR DESCRIPTION
**Pull request by GregoriBavaro**
- Installed Formik & Yup
- Implemented Formik Sample
- Integrated Yup for Schema validation
- Created Schema validation sample
- Implemented error handling with Formik to display errors 
- Implemented logic to reset form on submission on successful submit

### CODE REVIEW

**Comment by stoimenov95** [comment]
hey @GregoriBavaro, it would be great if we could separate the implementation from the sample form. We need the formik implementation to be reusable and easy to edit in order to add new rules for every use case we have in the future.
You can also write some tests for the sample form.

**Comment by GregoriBavaro** [comment]
Is this implementation for reusability ok? I will latter add some test, i have never written any test before for my code. What should i use. Google suggested  Jest and React Testing Library for writing tests.

File: components/reusable-components/formik/ReusableFormik.tsx [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+}: React.PropsWithChildren<Props<T>>) => {
+  const handleSubmit = (
+    values: FormValues,
+    { setSubmitting, resetForm }: FormikHelpers<FormValues>
+  ) => {
+    // Simulate form submission
+    setTimeout(() => {
+      setSubmitting(false);
+      resetForm(); // Reset the form values
+      console.log('values', values);
```
Comment (stoimenov95) [inline components/reusable-components/formik/ReusableFormik.tsx:31]: @GregoriBavaro do we need this log?

File: components/reusable-components/formik/ReusableFormik.tsx [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+              Submit
+            </button>
+          </Form>
+        )}
+      </Formik>
+    </div>
+  );
+};
+
+export default ReusableForm;
```
Comment (stoimenov95) [inline components/reusable-components/formik/ReusableFormik.tsx:60]: It's best if we can have the name as the filename.

File: components/reusable-components/formik/ReusableFormik.tsx (hunk 1 above)
Comment (GregoriBavaro) [reply components/reusable-components/formik/ReusableFormik.tsx:31]: Definitely not I forgot to remove it, I am removing it now, and removing the timeout as well

File: components/reusable-components/formik/ReusableFormik.tsx (hunk 2 above)
Comment (GregoriBavaro) [reply components/reusable-components/formik/ReusableFormik.tsx:60]: Missed it sorry, I will rename it

**Comment by GregoriBavaro** [comment]
> > Is this implementation for reusability ok? I will latter add some test, i have never written any test before for my code. What should i use. Google suggested Jest and React Testing Library for writing tests.
> 
> It looks much better now. Please look into the project details, we have a basic setup for Jest

Thank you @stoimenov95, i will write the test letter tonight

File: components/reusable-components/formik/ReusableFormik.tsx [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+  validationSchema,
+  children,
+}: React.PropsWithChildren<Props<T>>) => {
+  const handleSubmit = (
+    values: FormValues,
+    { setSubmitting, resetForm }: FormikHelpers<FormValues>
+  ) => {
+    setSubmitting(false);
+    resetForm(); // Reset the form values
+    // console.log('values', values);
```
Comment (stoimenov95) [inline components/reusable-components/formik/ReusableFormik.tsx:29]: you should remove it, not comment it

Comment (stoimenov95) [inline components/reusable-components/formik/ReusableFormik.tsx:29]: you should remove it, not comment it

Comment (GregoriBavaro) [reply components/reusable-components/formik/ReusableFormik.tsx]: Removed it

**Comment by edichoska** [comment]
@stoimenov95  I think we can merge this. Since we are not going to do the tests now.

**Comment by stoimenov95** [comment]
> @stoimenov95 I think we can merge this. Since we are not going to do the tests now.

I think @GregoriBavaro would like to try and write some tests for this. If not, feel free to merge it @GregoriBavaro.

**Comment by GregoriBavaro** [comment]
> > @stoimenov95 I think we can merge this. Since we are not going to do the tests now.
> 
> I think @GregoriBavaro would like to try and write some tests for this. If not, feel free to merge it @GregoriBavaro.

I am fine like this if there is no problem to merge it without test now and latter on the project i will practice writing tests anyway.

**Comment by edichoska** [comment]
@GregoriBavaro Please merge this

**Pull request merged**